api/enki.py

```diff
@@ -0,0 +1,554 @@
+        medium=Edition.BOOK_MEDIUM,
+        #series=series,
+        publisher=element["publisher"],
+        #imprint=imprint,
+        #published=publication_date,
+        primary_identifier=primary_identifier,
+        identifiers=identifiers,
+        #subjects=subjects,
+        contributors=contributors,
+        )
+        licenses_owned=element["availability"]["totalCopies"]
+        licenses_available=element["availability"]["availableCopies"]
+        hold=element["availability"]["onHold"]
+        formats = []
+        formats.append(FormatData(content_type=Representation.EPUB_MEDIA_TYPE, drm_scheme=DeliveryMechanism.ADOBE_DRM))
+
+        circulationdata = CirculationData(
+            data_source=DataSource.ENKI,
+            primary_identifier=primary_identifier,
+            formats=formats,
+            licenses_owned = int(licenses_owned),
+            licenses_available = int(licenses_available),
+            patrons_in_hold_queue = int(hold)
+        )
+
+        metadata.circulation = circulationdata
+        return metadata
+
+
+    def process_one(self, element):
+        if self.include_bibliographic:
+            bibliographic = self.extract_bibliographic(element)
+        else:
+            bibliographic = None
+
+        availability = None
+        if self.include_availability:
+            availability = bibliographic.circulation
+
+        return bibliographic, availability
+
+class EnkiImport(Monitor):
+    """Import Enki titles.
+    """
+
+    VERY_LONG_AGO = datetime.datetime(1970, 1, 1)
+    FIVE_MINUTES = datetime.timedelta(minutes=5)
+
+    def __init__(self, _db, name="Enki Import",
+                 interval_seconds=60, batch_size=50, api=None):
+	super(EnkiImport, self).__init__(
+            _db, name, interval_seconds=interval_seconds,
+            default_start_time = self.VERY_LONG_AGO
+        )
+        self.batch_size = batch_size
+        self.api = api or EnkiAPI.from_environment(self._db)
+        self.bibliographic_coverage_provider = (
+            EnkiBibliographicCoverageProvider(self._db, enki_api=api)
+        )
+
+    def run_once(self, start, cutoff):
+        # Give us five minutes of overlap because it's very important
+        # we don't miss anything.
+        since = start-self.FIVE_MINUTES
+        id_start = 0
+        while True:
+            availability = self.api.availability(since=since, strt=id_start, qty=self.batch_size)
+            if availability.status_code != 200:
+                self.log.error(
+                    "Could not contact Enki server for content availability. Status: %d",
+                    availability.status_code
+                )
+            content = availability.content
+            count = 0
+            for bibliographic, circulation in BibliographicParser().process_all(content):
+                self.process_book(bibliographic, circulation)
+                count += 1
+            if count == 0:
+                break
```

Comment:
For my own understanding, does it make a difference? If process_book never runs, is there anything to commit? I figured I'd save on an extra commit operation if it wasn't going to do anything.